lib/matplotlib/ticker.py

```diff
@@ -1292,6 +1302,128 @@ def convert_to_pct(self, x):
         return 100.0 * (x / self.xmax)
 
 
+class IntegerFormatter(StrMethodFormatter):
+    """
+    Formats the input as an integer with a specified minimum number of
+    digits.
+
+    ``unsigned`` determines if negative numbers are printed with a minus
+    sign or if their binary representation is truncated to ``bits``
+    digits and treated as a positive number.
+
+    ``bits`` the number of bits that actually represent the number. It
+    is only used in ``unsigned`` mode for negative numbers. The default
+    is ``(-x).bit_length() + 1``.
+
+    This formatter is very similar to :class:`StrMethodFormatter`.
+    However, the integer formats (``b``, ``o``, ``d``, ``n``, ``x``,
+    ``X``) can not be used with with floating point inputs that normally
+    represent tick locations. This formatter converts inputs to an
+    integer before passing them to `StrMethodFormatter` for actual
+    conversion. Since the input is truncated to an integer, this
+    `Formatter` is most effective with a `MaxNLocator` configured with
+    ``integer=True``.
+    """
+    def __init__(self, fmt, unsigned=False, bits=None):
+        self._unsigned = bool(unsigned)
+        if bits is not None:
+            bits = int(bits)
+        self._bits = bits
+        super(IntegerFormatter, self).__init__(fmt)
+
+    def transform(self, x):
+        """
+        Ensures that `x` is an integer.
+
+        If `self.unsigned` is `True`, then the integer is treated as a
+        positive number of size `self.bits`. If `self.bits` is `None`,
+        the number of bits is taken as `(-x).bit_length() - 1`.
+
+        Returns the transformed number.
+        """
+        x = int(x)
+        if x < 0 and self._unsigned:
+            bits = (-x).bit_length() if self._bits is None else self._bits
+            x = x & ((1 << bits) - 1)
+        return x
+
+    def __call__(self, x, pos=None):
+        return super(IntegerFormatter, self).__call__(x=self.transform(x),
+                                                      pos=pos)
+
+
+class LinearScaleFormatter(Formatter):
+    """
+    Displays labels that are scaled and offset versions of the actual
+    axis values.
+
+    This formatter can use any other formatter to actually render the
+    ticks.
+
+    inRef: number or 2-element iterable
```

Comment:
The last input to this is already a formatter instance, maybe take in a `Norm` instance as the input here?

If not, I have a slight preference to 4 inputs instead of 2 of mixed type.